PukingPredator/Assets/Scripts/Eating.cs

```diff
@@ -65,26 +71,46 @@ private void GameInput_Eat()
         {
             // Get the GameObject that was hit
             GameObject hitObject = hit.collider.gameObject;
-
-            var consumableData = hitObject.GetComponent<Consumable>();
-            if (consumableData == null || !consumableData.isConsumable) { return; }
-
-            inventory.PushItem(consumableData);
-            consumableData.SetState(ItemState.beingConsumed);
+            Process_Consumption(hitObject);
         }
-    }
+        else if (Physics.SphereCast(ray, radius: 0.2f, out hit, maxDistance: 2f, layerMask: dashLayers))
+        {
+            // Dashes if the object is far
+            GameObject hitObject = hit.collider.gameObject;
+            Process_Consumption(hitObject);
+
+            Player player = player = this.GetComponent<Player>();

```

Comment:
why is there two = here?
also, is it possible to run the GetComponent in the Start() function instead of every time this function is called?